forest/main.py

```diff
@@ -183,12 +187,16 @@ def is_image(renderer):
         bokeh.layouts.column(dropdown))
 
 
-    navigator = navigate.Navigator({key: dataset.navigator()
-                                    for key, dataset in datasets.items()})
+    # Add optional sub-navigators
+    sub_navigators = {
+        key: dataset.navigator() for key, dataset in datasets.items()
+        if hasattr(dataset, "navigator")
+    }
+    navigator = navigate.Navigator(sub_navigators)
 
     # Pre-select menu choices (if any)
     initial_state = {}
-    for _, pattern in config.patterns:
+    for pattern, _ in sub_navigators.items():
```

Comment:
I am a bit confused here. The keys in `sub_navigators` are now the patterns for db.initial_date, but it looks like they are in fact Navigators returned by `dataset.navigator()`. Can't quite tell why this all works!

Reply:
@twsearle yes this definitely needs to change in a future PR, the keys for the Navigator should be a unique ID to reference sub-navigators

Reply:
@twsearle also, perhaps it's time for `db.initial_state()` to move somewhere else?